### PR TITLE
Refactor/ibs 76 migrate mark student

### PR DESCRIPTION
**Pull request by ANJEW**
## Description

- Refactored endpoints for get mark as student.

## Changes Made

- Refactored following endpoint: backend/module/mark/student/get.js
- Added Criteria id FK to Mark table: backend/migrations/20231008140230-create-mark.js
- Removed Task.task FK from Criteria table:  backend/migrations/20231008140126-create-criteria.js
- Refactored helper functions relating endpoint: backend/utilities/helpers.js

## Test Guidelines

1. cd IBS/backend
2. npm test mark_get_student

### For PR author:
- [ ] I have reviewed my code.
- [ ] I have tested my changes locally.
- [ ] I have added appropriate comments to the code.
- [ ] I have ensured code style and formatting are consistent.
- [ ] I have included unit tests or integration tests where applicable.
- [ ] I have considered edge cases and handled them appropriately.
- [ ] I have celebrated my accomplishments 🎉

### For PR reviewer:
- [ ] I have reviewed the code changes.
- [ ] I have tested the changes locally.
- [ ] I have verified that comments are clear and helpful.
- [ ] I have ensured code style and formatting align with project standards.
- [ ] I have examined any included tests and their coverage.
- [ ] I have considered potential edge cases and provided feedback on them.

### CODE REVIEW

File: backend/module/mark/student/get.js [hunk 1]
```diff
@@ -1,38 +1,47 @@
...
-router.get("/", (req, res) => {
-    if (req.query["total"] === true || req.query["total"] === "true") {
-        var total = true;
-    } else {
-        var total = false;
-    }
+router.get("/", async (req, res) => {
+    let total;
+    try {
+        total = req.body["total"] === true || req.body["total"] === "true";
```
Comment (DomiVesalius) [inline backend/module/mark/student/get.js:11]: Unused variable

File: backend/module/mark/student/get.js [hunk 2]
```diff
@@ -1,38 +1,47 @@
...
+                where: {username: res.locals.username, hidden: false},
+                attributes: [
+                    "username",
+                    "task_name",
+                    [sequelize.fn("SUM", sequelize.col("mark")), "sum"],
+                ],
+                group: ["username", "task_name"],
+            });
+            
+            marks = await helpers.format_marks_all_tasks(marksData, res.locals.course_id, task);
```
Comment (DomiVesalius) [inline backend/module/mark/student/get.js:28]: This helper only seems to take in 2 argument but 3 arguments seem to be passed here. Not sure why. Please clarify

https://github.com/Kianoosh76/IBS/blob/410ad3d56cd7c3a82a734097467d3393803705a9/backend/utilities/helpers.js#L643

File: backend/module/mark/student/get.js [hunk 3]
```diff
@@ -1,38 +1,47 @@
...
+
+        } else {
+            // Retrieve marks for a specific task for the student
+            marksData = await Mark.findAll({
+                where: {username: res.locals.username, hidden: false},
+                order: [["criteria_id", "ASC"]],
+            });
```
Comment (DomiVesalius) [inline backend/module/mark/student/get.js:35]: This does not retrieve marks for a specific task. It retrieves marks for every task that the user is associated with which is not the behavior that the original endpoint had.

File: backend/seeders/sd_staff_mark.js
Comment (DomiVesalius) [inline backend/seeders/sd_staff_mark.js:1]: Empty file committed.

**Review by DomiVesalius — CHANGES_REQUESTED** [review]
Left a few comments.

Mostly good however there are 2 glaring issues:
1. SQL -> Sequelize translation is not exactly right.
2. Tests pass however I get this log output which isn't normal and needs to be addressed.

After running `npm test mark_get_student`

image
